framework/build.mk

```diff
@@ -57,6 +57,16 @@ ifneq (,$(findstring mpi,$(cxx_compiler)))
 	cxx_compiler = $(shell $(libmesh_CXX) -show)
 endif
 
+# Check that we have Asio installed.
+ASIO_FILE := $(MOOSE_DIR)/framework/contrib/asio/include/asio.hpp
+ifneq ("$(wildcard $(ASIO_FILE))","")
+# If it is installed, make sure this is a 
+# build using C++11, if so, then turn on Asio Networking
+ifeq (,$(findstring "-std=gnu++11",$(libmesh_CXXFLAGS)))
```

Comment:
@permcody @jwpeterson So you guys are saying that checking for -std=gnu+11 won't work for some of the other compilers? How do you check in the Makefile for the LIBMESH_HAVE_CXX11 constant?